platine-management-service/src/main/java/fr/insee/survey/datacollectionmanagement/contact/service/impl/ContactServiceImpl.java

```diff
@@ -56,6 +63,39 @@ public void deleteContact(String identifier) {
         contactRepository.deleteById(identifier);
     }
 
+    @Override
+    @Transactional
+    public Contact updateOrCreateContact(String id, ContactDto contactDto, JsonNode payload) {
+        // Vérifier si le contact existe
+        Optional<Contact> existingContact = contactRepository.findById(id);
+
+        if (existingContact.isPresent()) {
+            // Mise à jour du contact existant
+            Contact contact = convertToEntity(contactDto);
+
+            if (contactDto.getAddress() != null) {
+                contact.setAddress(addressService.convertToEntity(contactDto.getAddress()));
+            }
+
+            return updateContactAddressEvent(contact, payload);
+
+        } else {
+            // Création d'un nouveau contact
+            Contact newContact = convertToEntityNewContact(contactDto);
+
+            if (contactDto.getAddress() != null) {
+                newContact.setAddress(addressService.convertToEntity(contactDto.getAddress()));
+            }
+
+            Contact createdContact = createContactAddressEvent(newContact, payload);
+
+            // Créer une vue pour le nouveau contact
+            viewService.createView(id, null, null);
```

Comment:
refactoring name to understand we're talking about contacts ? Why not using a real view instead of a table ?